### PR TITLE
Improving documentation - Creating Vitepress documentation

### PR DESCRIPTION
**Pull request by linspw**
Hello, my name is Jessé, I really liked this library, it proved to be much easier and safer than other implementations.

I want to contribute to the project to grow and improve.

In this PR, I'm creating Vitepress, official vue 3 documentation tool

It's simple, beautiful and lets you use live examples.

If they accept this implementation, I will go after a way to deploy it (in another PR).

Thank you very much in advance!

### Screenshot:
image

### Changes:
- Add Vitepress
- Everthing about Vitepress is created in `/docs`
- `/img` moved to `/docs/img` (made a big difference in PR).

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+
+features:
+  - title: Simple API
+    details: With a simple API, you can create complex and scalable solutions.
+  - title: Easy Installation
+    details: Quick to install and supports SSR, Vue 3 and 2.
+  - title: Easy to use
+    details: Tools are very simple compared to other DnD APIs.
+---
```
Comment (linspw) [inline docs/index.md:25]: @nicholaszuccarelli

**Comment by linspw** [comment]
Hello Nicholas, thank you for your attention.

Vitepress is very simple, it practically uses MD files to present the documentation.

And after the build process is done, the website it generates statically is ready to deploy.

And yes, we can use GitHub Pages, it will work just fine.

https://vitepress.dev/guide/deploy#github-pages

About the documentation I transcribed from `README.md`, just add the features section of the Vitepress main page, feel free to change it for what you think is best. https://github.com/rlemaigre/Easy-DnD/pull/139/files#r1171449562

About Vite, it is used internally by Vitepress and does not impact this PR in the library itself.

I'm going to open another PR with the GitHub Pages deploy.

Many thanks again Nicholas!

Jessé

**Comment by linspw** [comment]
PR with deploy:
https://github.com/linspw/Easy-DnD/pull/1

Link with documentation:
https://linspw.github.io/Easy-DnD/

**Comment by rlemaigre** [comment]
This so cool.

Very rewarding to see that my little project still generates some interest and has been picked up by people who contribute and maintain it now that I don't really use it myself anymore (it's been some time since I last needed drag and drop in a project of mine).

Many thanks.

**Comment by nicholaszuccarelli** [comment]
Hey Jesse.
Sorry I got extremely busy and just forgot about this.
I'll take a look at it today :)

**Comment by nicholaszuccarelli** [comment]
@rlemaigre I have set it up and pushed to master.
One step I need you to do from Vitepress:

`In your repository's Settings under Pages menu item, select GitHub Actions in Build and deployment's Source.`
https://vitepress.dev/guide/deploy#github-pages

I've set up the config files etc, just need you to enable Github Pages in the project settings (I don't have access to it).

**Comment by nicholaszuccarelli** [comment]
@linspw If you happen to have time and want to contribute more, some updated codepens which use Vue3 would be amazing.
The current codepens are very out of date.

**Comment by rlemaigre** [comment]
> @rlemaigre I have set it up and pushed to master. One step I need you to do from Vitepress:
> 
> `In your repository's Settings under Pages menu item, select GitHub Actions in Build and deployment's Source.` https://vitepress.dev/guide/deploy#github-pages
> 
> I've set up the config files etc, just need you to enable Github Pages in the project settings (I don't have access to it).

I've enabled Github Pages with Github Actions as you requested. Let me know if there is anything more I need to do. The Pages panel looks like this now :

image

**Comment by nicholaszuccarelli** [comment]
I think that is it.
I'm not at my PC right now. Are u able to make a blank commit on master?
It should automatically then create the github pages website.

**Comment by nicholaszuccarelli** [comment]
I got it to deploy but its still glitchy. Will get it working properly tomorrow :)

**Comment by rlemaigre** [comment]
Many thanks to both of you :)

**Comment by linspw** [comment]
I thank you @nicholaszuccarelli @rlemaigre, I'll see if I can bring more improvements in my free time

**Pull request merged**